src/python/pants/auth/cookies.py

```diff
@@ -0,0 +1,52 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import os
+
+from six.moves.http_cookiejar import LWPCookieJar
+
+from pants.subsystem.subsystem import Subsystem
+from pants.util.dirutil import safe_mkdir_for
+
+
+class BasicAuthException(Exception):
+  pass
+
+
+class Cookies(Subsystem):
+  options_scope = 'cookies'
+
+  @classmethod
+  def register_options(cls, register):
+    super(Cookies, cls).register_options(register)
+    register('--path', advanced=True, default='~/.pants.cookies',
+             help='Path to file that stores persistent cookies.')
+
+  def update(self, cookies):
+    """Add specified cookies to our cookie jar, and persists it.
+
+    :param cookies: Any iterable that yields http.cookiejar.Cookie instances, such as a CookieJar.
+    """
+    cookie_jar = self.get_cookie_jar()
+    for cookie in cookies:
+      cookie_jar.set_cookie(cookie)
+    cookie_jar.save()
```

Comment:
I think the `save`s will need `OwnerPrintingInterProcessFileLock` since the the default (and sensible) path is global. The `LWPCookieJar` docs don't say anything about multi-process safety and a quick look at the 2.7 source says its not. Towards this end, maybe it makes more sense to default the path to `~/.cache/pants/auth/...` or `~/.config/pants/auth/...` to give a more scoped location for the lock file. Not fully convinced of this latter bit.

Reply:
Hmm, you are correct. 

It's slightly weird, now that I think of it, that logins will be shared across all repos. However putting the cookies under .pants.d seems wrong, because we probably don't want a clean to log you out. Or do we? Maybe we do. I can't make up my mind on this.  Thoughts?

If we put the cookies under .pants.d then the regular pants lock will suffice.

Reply:
No - I think sharing across repos makes loads of sense. It might not for the services you have in mind ;), but - in general, I should only have to log into a given provider url once. If two repos both use that same provider url, don't make me log in again.